test/count-zero-bits.ll

```diff
@@ -0,0 +1,62 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -spirv-text -o - | FileCheck %s
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: spirv-val %t.spv
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir64-unknown-unknown"
+
+; CHECK: ExtInstImport [[extinst_id:[0-9]+]] "OpenCL.std"
+
+; CHECK: Function
+; CHECK: 6 ExtInst {{[0-9]+}} {{[0-9]+}} [[extinst_id]] clz
+; CHECK: FunctionEnd
+
+; Function Attrs: nounwind readnone
+define spir_func i32 @TestClz(i32 %x) local_unnamed_addr #0 {
+entry:
+  %0 = tail call i32 @llvm.ctlz.i32(i32 %x, i1 true)
+  ret i32 %0
+}
+
+; CHECK: Function
+; CHECK: 6 ExtInst {{[0-9]+}} {{[0-9]+}} [[extinst_id]] ctz
+; CHECK: FunctionEnd
+
+; Function Attrs: nounwind readnone
+define spir_func i32 @TestCtz(i32 %x) local_unnamed_addr #0 {
+entry:
+  %0 = tail call i32 @llvm.cttz.i32(i32 %x, i1 true)
+  ret i32 %0
+}
+
+; CHECK: Function
+; CHECK: 6 ExtInst {{[0-9]+}} {{[0-9]+}} [[extinst_id]] ctz
+; CHECK: FunctionEnd
+
+; Function Attrs: nounwind readnone
+define spir_func <4 x i32> @TestCtzVec(<4 x i32> %x) local_unnamed_addr #0 {
+entry:
+  %0 = tail call <4 x i32> @llvm.cttz.v4i32(<4 x i32> %x, i1 true)
+  ret <4 x i32> %0
+}
+
+; Function Attrs: nounwind readnone speculatable willreturn
+declare i32 @llvm.ctlz.i32(i32, i1 immarg) #1
```

Comment:
LGTM, but I wonder if we should handle `immarg` as `{{(immarg)?}}` to reduce the diff between branches, or whether it would be better not.

Reply:
It's not that trivial to make `immarg` optional in the LLVM IR input file: I guess we'd have to run some preprocessing step.  I'm inclined to say that avoiding a small diff between previous release branches doesn't justify adding such clutter on master.

Reply:
Good point!  If we had a _very_ lightweight and conventional way of doing the preprocessing, that might be a different matter, but I think anything we do will be too intrusive and also too surprising to be justifiable.